### PR TITLE
ADCM-6117: Implement Component object

### PR DESCRIPTION
**Pull request by DanBalalan**

### CODE REVIEW

File: adcm_aio_client/core/objects/cm.py [hunk 1]
```diff
@@ -130,9 +133,46 @@ class ServicesNode(PaginatedChildAccessor[Cluster, Service, None]):
     class_type = Service
 
 
-class Component(InteractiveChildObject[Service]):
+class Component(
+    WithStatus, WithActions, WithConfig, WithActionHostGroups, WithConfigGroups, InteractiveChildObject[Service]
+):
+    PATH_PREFIX = "components"
```
Comment (Sealwing) [inline adcm_aio_client/core/objects/cm.py:139]: 🐗 

technically it's not prefix 🧌 , it's infix

**Review by Sealwing — CHANGES_REQUESTED** [review]

File: adcm_aio_client/core/objects/cm.py [hunk 2]
```diff
@@ -130,9 +133,46 @@ class ServicesNode(PaginatedChildAccessor[Cluster, Service, None]):
...
+
+        raise NotFoundError
+
+    @cached_property
+    def service(self: Self) -> Service:
+        return self._parent
+
+    @cached_property
+    def cluster(self: Self) -> Cluster:
+        return self._parent._parent
```
[the remark this answers is not in the document]
Comment (DanBalalan) [reply adcm_aio_client/core/objects/cm.py]: refactored

File: adcm_aio_client/core/objects/cm.py [hunk 3]
```diff
@@ -130,9 +133,46 @@ class ServicesNode(PaginatedChildAccessor[Cluster, Service, None]):
...
+
+    @cached_property
+    def hosts(self: Self) -> "HostsInClusterNode":
```
[the remark this answers is not in the document]
Comment (DanBalalan) [reply adcm_aio_client/core/objects/cm.py:167]: will be changed in ADCM-6133

File: pyproject.toml [hunk 4]
```diff
@@ -9,6 +9,7 @@ readme = "README.md"
 [tool.poetry.dependencies]
 python = "^3.12"
 httpx = "^0.27.2"
+asyncstdlib = "^3.13.0"
```
[the remark this answers is not in the document]
Comment (DanBalalan) [reply pyproject.toml:12]: `async_cached_property` works with async callables only, so we need both `cached_property` versions

**Review by Sealwing — DISMISSED** [review]

**Review by Starovoitov — DISMISSED** [review]

**Review by Sealwing — DISMISSED** [review]

**Review by Starovoitov — DISMISSED** [review]

**Review by Sealwing — DISMISSED** [review]

**Review by Sealwing — APPROVED** [review]

**Review by Starovoitov — APPROVED** [review]

**Pull request merged**